src/screens/authentication/components/Auth/Form/index.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import {
+  KeyboardAvoidingView,
+  TextInput,
+  View
+} from 'react-native'
+import Icon from 'react-native-vector-icons/FontAwesome5'
+import { Formik } from 'formik'
+import { values } from 'lodash'
+import { Body, Description, GrayButton, PrimaryButton, Button } from '~/components'
+import AuthenticatingIndicator from '../../AuthenticatingIndicator'
+import styles, { PLACEHOLDER_COLOR, PASSWORD_TOGGLE_ICON_SIZE } from './styles'
+
+const Error = ({ error }) => (
+  error && (
+    <Body style={styles.error}>
+  Credenciales no aceptadas
+    </Body>
+  )
+)
+
+const SubmitButton = ({ onPress, thereAreEmptyValues, error, disabled, children }) => {
+  if (thereAreEmptyValues || error) {
+    return (
+      <GrayButton style={styles.submitButton}>
+        <Description white>
+          {children}
+        </Description>
+      </GrayButton>
+    )
+  } else {
+    return (
+      <PrimaryButton
+        onPress={onPress}
+        style={styles.submitButton}
+        disabled={disabled}
+      >
+        <Description white>
+          {children}
+        </Description>
+      </PrimaryButton>
+    )
+  }
+}
+
+const PasswordVisibleToggle = ({ onPress, isVisible }) => {
+  return (
+    <Button
+      onPress={onPress}
+      style={styles.togglePasswordButton}
+    >
+      <Icon
+        type='font-awesome'
+        name={isVisible ? 'eye-slash' : 'eye'}
+        size={PASSWORD_TOGGLE_ICON_SIZE}
+        color={PLACEHOLDER_COLOR}
+      />
+    </Button>
+  )
+}
+
+const areFormValuesEmpty = (formValues) => {
+  const emptyValues = values(formValues).filter(element => element === '')
+  return emptyValues.length !== 0
+}
+
+const Form = ({
+  initialValues,
+  handleSubmit,
+  schema,
+  error,
+  authenticating,
+  credentialText,
+  submitText
+}) => {
```

Comment:
While this is true, `schema` is only used in the case of `SignUp`, as it would disturb the user in the case of the Login (text is displayed in case of missing fields), there is also some references to values in the handling method. I can move the `initialValues` and `schema` inside the component and provide a prop `doNotValidate`, but I think the artificial coupling will continue to exist due to code and style reuse.